src/utils.js

```diff
@@ -423,6 +474,21 @@ export function retrieveSchema(schema, definitions = {}, formData = {}) {
   }
 }
 
+export function retrieveSchema(schema, definitions = {}, formData = {}) {
+  const resolvedSchema = resolveSchema(schema, definitions, formData);
+  const hasAdditionalProperties =
+    resolvedSchema.hasOwnProperty("additionalProperties") &&
+    resolvedSchema.hasAdditionalProperties !== false;
```

Comment:
What is the `hasAdditionalProperties` property? Is this maybe supposed to be `additionalProperties`?